### PR TITLE
fix: Add parameter validation for removeObject and getObject

### LINKED ISSUE #651: Argument validation should be more strict.
Hi!
In this method:
``` java
public void removeObject(String bucketName, String objectName)
```
If objectName is null or empty(due to some bad coding),this method will delete the whole bucket,which is very dangerous,even removeBucket would not recursively delete objects.

The api description should be more precise,and do the argument validation strictly,if there is a  invalid argument , an IllegalArgumentException would be better.

There is other  method like `getObject(String bucketName, String objectName)`,if objectName is null or empty,this method would act as ListBucketResult with no prefix.

### 3.0.8

### PR DESCRIPTION
**Pull request by nitisht**
fixes #651

### CODE REVIEW

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -1505,8 +1505,15 @@ public String getObjectUrl(String bucketName, String objectName)
...
-           InternalException,
-           InvalidArgumentException {
+           InternalException, InvalidArgumentException {
+    if ((bucketName == null) || (bucketName.isEmpty())) {
+      throw new InvalidArgumentException("bucket name cannot be empty");
+    }
+    
+    if ((objectName == null) || (objectName.isEmpty())) {
+      throw new InvalidArgumentException("object name cannot be empty");
+    }
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: Add this check at line #1849 and #1581

Comment (nitisht) [reply api/src/main/java/io/minio/MinioClient.java]: updated @balamurugana

Comment (balamurugana) [reply api/src/main/java/io/minio/MinioClient.java]: If so, why this check is here?

File: api/src/main/java/io/minio/MinioClient.java [hunk 2]
```diff
@@ -1586,6 +1593,14 @@ public InputStream getObject(String bucketName, String objectName, long offset,
       throw new InvalidArgumentException("length should be greater than zero");
     }
 
+    if ((bucketName == null) || (bucketName.isEmpty())) {
+      throw new InvalidArgumentException("bucket name cannot be empty");
+    }
+    
+    if ((objectName == null) || (objectName.isEmpty())) {
+      throw new InvalidArgumentException("object name cannot be empty");
+    }
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: Do this validation first.  See the order of method arguments.

**Comment by nitisht** [comment]
I get your point now @balamurugana . Updated the PR to add checks only for final `getObject` and `removeObject` methods.  Removed the checks from wrapper methods

File: api/src/main/java/io/minio/MinioClient.java [hunk 3]
```diff
@@ -1576,16 +1575,23 @@ public InputStream getObject(String bucketName, String objectName, long offset)
...
+      
     if (offset < 0) {
       throw new InvalidArgumentException("offset should be zero or greater");
     }
 
     if (length != null && length <= 0) {
       throw new InvalidArgumentException("length should be greater than zero");
     }
-
+
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: fix this whitespace

Comment (nitisht) [reply api/src/main/java/io/minio/MinioClient.java]: done

**Pull request merged**